src/ui/lp-render.ts

```diff
@@ -0,0 +1,245 @@
+/*
+* inspired and adapted from https://github.com/artisticat1/obsidian-latex-suite/blob/main/src/conceal.ts
+*
+* The original work is MIT-licensed.
+*
+* MIT License
+*
+* Copyright (c) 2022 artisticat1
+*
+* Permission is hereby granted, free of charge, to any person obtaining a copy
+* of this software and associated documentation files (the "Software"), to deal
+* in the Software without restriction, including without limitation the rights
+* to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+* copies of the Software, and to permit persons to whom the Software is
+* furnished to do so, subject to the following conditions:
+*
+* The above copyright notice and this permission notice shall be included in all
+* copies or substantial portions of the Software.
+*
+* THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+* IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+* FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+* AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+* LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+* OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+* SOFTWARE.
+*
+* */
+
+
+import {EditorView, ViewUpdate, Decoration, ViewPlugin, DecorationSet, WidgetType} from "@codemirror/view";
+import { EditorSelection, Range } from "@codemirror/state";
+import { syntaxTree } from "@codemirror/language";
+import {DataviewSettings} from "../settings";
+import { FullIndex } from "../data-index";
+import {Component, editorLivePreviewField, TFile} from "obsidian";
+import {DataviewApi} from "../api/plugin-api";
+import {renderMinimalDate, renderMinimalDuration, tryOrPropogate} from "../util/normalize";
+import {parseField} from "../expression/parse";
+import {executeInline} from "../query/engine";
+import {currentLocale} from "../util/locale";
+import {Literal, Values} from "../data-model/value";
+import {DataviewInlineApi} from "../api/inline-api";
+
+function selectionAndRangeOverlap(selection: EditorSelection, rangeFrom:
+    number, rangeTo: number) {
+
+    for (const range of selection.ranges) {
+        if ((range.from <= rangeTo) && (range.to) >= rangeFrom) {
+            return true;
+        }
+    }
+
+    return false;
+}
+
+
+class InlineWidget extends WidgetType {
+    el: HTMLElement;
+
+    constructor(readonly settings: DataviewSettings, readonly currentFile: TFile,
+                readonly cssClasses: string[], readonly rawQuery: string, readonly markdown?: string | Literal) {
+        super();
+        this.el = createSpan({
+            text: '',
+            cls: ['dataview', 'dataview-inline']
+        })
```

Comment:
This HTML element gets created also when it won't be used (if `eq` says it's the same). Could that cause a memory leak?